tests/core/string/test_string.h

```diff
@@ -1274,10 +1274,28 @@ TEST_CASE("[String] is_lowercase") {
 }
 
 TEST_CASE("[String] match") {
+	// Base cases.
+	CHECK(String("img1.png").match("img1.png"));
+	CHECK(!String("iMG1.png").match("img1.png"));
+	CHECK(String("iMG1.png").matchn("img1.png"));
+
+	// Single-char wildcard cases.
+	CHECK(String("img2.png").matchn("img?.png"));
+	CHECK(!String("img23.png").matchn("img?.png"));
```

Comment:
While we're at it, we can add a few more test cases:

```suggestion
	// Single-char wildcard cases.
	CHECK(String("img2.png").matchn("img?.png"));
	CHECK(!String("img23.png").matchn("img?.png"));
	CHECK(String("img12345.png").matchn("img?????.???"));
	CHECK(!String("img123456.png").matchn("img?????.???"));
```